code/segmentedtickerplant/datastripe.q

```diff
@@ -30,29 +26,33 @@ initdatastripe:{
 
 // Function to map the where clause from config table extracted by .stpps.segmentfilter function to tablename
 // Allows use of ` as argument for tables
-filtermap:{[tabs;id] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;id]}
+filtermap:{[tabs;id] if[tabs~`;tabs:.stpps.t]; ((),tabs)!.stpps.segmentfilter\:[(),tabs;id]};
 
 // Find where clause from config tables
 segmentfilter:{[tbl;segid]
-     wcref:first exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid;
-     .stpps.segmentfiltermap[wcref]
+       filter:first exec filter from .stpps.segmentconfig where segmentID=segid,table=tbl;
+       $[0=count filter;filter:"";filter]
      };
 
 // Subscribe to particular segment using segmentID based on .u.sub
+// Uses .stpps.segmentconfig to handle subscriptions for each table
 subsegment:{[tbl;segid];
      //tablename and segmentid used to get filters
      if[tbl~`;:.z.s[;segid] each .stpps.t];
      if[not tbl in .stpps.t;
           .lg.e[`sub;m:"Table ",string[tbl]," not in list of stp pub/sub tables"];
-          :();
+          :(tbl;`err`msg!(`table;m));
      ];
      filter:segmentfilter[tbl;segid];
-     if[filter~"";
-          .lg.e[`sub;m:"Incorrect pairing of table ",string[tbl]," and segmentID ",string[segid]," not found in .stpps.segmentconfig"];
-          :();
-     ];
-     
-     .ps.subtablefiltered[string[tbl];filter;""]
+     $[tbl in .stpps.fullsubscriptions;[.stpps.suball[tbl]];
+     [$[tbl in .stpps.ignoredtables;
+                    [.lg.o[`sub;m:"Table ",string[tbl]," is in ignoredtables.csv and will not be subscribed to"];
+                    :();];
+       [$[filter~"";             
+                    [.lg.e[`sub;m:"Incorrect pairing of table ",string[tbl]," and segmentID ",string[segid]," not found in .stpps.segmentconfig"];
+                    :(tbl;`err`msg!(`segmentid;m));]
+                    ;
+                    [.ps.subtablefiltered[string[tbl];filter;""]]]]]]]
```

Comment:
I think this nested conditional is a bit confusing and hard to follow - could it be either re-written as separate `if` statements, or re-formatting in some way?

e.g. I think something like the following would have the same functionality but is easier to follow IMO using early exits so that each subsequent if statement only gets executed if the previous ones were false:

```suggestion
     if[tbl in .stpps.fullsubscriptions;:.stpps.suball[tbl]];
     if[tbl in .stpps.ignoredtables;
        .lg.o[`sub;m:"Table ",string[tbl]," is in ignoredtables.csv and will not be subscribed to"];
        :();
     ];
     if[filter~"";          
        .lg.e[`sub;m:"Incorrect pairing of table ",string[tbl]," and segmentID ",string[segid]," not found in .stpps.segmentconfig"];
        :(tbl;`err`msg!(`segmentid;m));
     ];
     .ps.subtablefiltered[string[tbl];filter;""]
```

Reply:
Did not know it could be done like that and it all works Thanks